PR for To do List

smcommits opened the pull request:
# Description

In this PR, We have implemented a simple to do list application. The goal  was to use javascript to create an application where a user can create new projects, add tasks for each project, edit the project and tasks, and delete the tasks. 

## Page Contents 
- Project List 
- Task List
- Controll Buttons
- Create and edit form 


## Javascript
We have used the ES6 version of Javascript. We have used the MVC software design to divide the code and responsibility. 

## Technologies Used
- Webpack for bundling the code 
- NPM for dependencies and script handling
- ESLINT for lining Javascript
- Sylelint for lining CSS

sinansevgi committed:
Basic project structure

Signed-off-by: Sinan Sevgi <sinansevgi@users.noreply.github.com>

smcommits committed:
Add Project, Todo, and LocalStorage Class

- Added Project class to encapsulate the logic related to the creation
  and management of product. The project class contains a static
  variable to hold all the projects.

- Added Todo class to encapsulate the logic related to the creation and
  management of Todo tasks. The Todo Class has the operations

sinansevgi committed:
Add html template

Signed-off-by: Sinan Sevgi <sinansevgi@users.noreply.github.com>

smcommits committed:
Create Controller For Project

Created project controller to establish communication between project view and project model. The point
of project conroller is abstraction.

sinansevgi committed:
Add controller and view for tasks.

smcommits committed:
Add functionality to create new task

Added the functionality to creat new tasks. Every todo list render gets a data-attribute that corresponds to the projects index in project-list array.
The data attribute is used to make sure that the task created or manipulated is addressing the correct project

sinansevgi committed:
Fix view and modules

smcommits committed:
Refactor DOM operations

Created a helper module to handle the repeated DOM operations. It
contains fucntion for setting multiple attributes, appending multiple
child, and creatingDomElement. The function creatingDomElement takes
attributes as a key value object and uses setAttribute to assign the
attributes.

smcommits committed:
fix conflict

sinansevgi committed:
Add remove project

smcommits committed:
Add functionality to change status, and edit task

Added the functionality to change the status. The task radio button listens to click event
and toggles the state of any task based on the boolean value of the checkbox. It also toggles the display on
completed task badge

Added form trigger to create an edit form for a task.

sinansevgi committed:
Add functionality for remove task and completed edit task.

- Updated edit form functionality so it can update related dom element.

- Add seed function for default project.

smcommits committed:
Update Reamde, Add description to default task

Updated reamde to contain all the basic informtation about the project,
how to get started, how to install dependencies, and run the project.

Add a description to default task which dictates how to run the
application.

smcommits committed:
Add title for page.

Forison reviewed (changes requested):
## Status::Change required❌

# 🤗 team, 

#### You have done a great job on this one, I Iike the way the project has been organized, the code base is very clean and readable. The UI is also amazing. Here is one requirement we have to address and other optional requirement that could even make the app better😇 .

* [ ] Let us make sure the necessary run common commands like `build`, `development` should be in `package.json` inside \`scripts'. Here is a basic example ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/javascript/todo_list/scripts_requirement.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59622602/))
- Let add this ``"build": "webpack --mode production",`` to our script.

**[Optional]** 
* [ ] The forms used to create a project or a todo should have validations, you should not be able to create an empty project/todo, you can simply add the required attribute to your input fields.
* [ ] Currently we can create duplicate projects name, I think porject it would be better if project is unique, that is just my view.

#### That is it, kindly make the changes and submit a re-review

#### Happy coding🙂
Addo Forison

Forison commented on package.json:
```diff
@@ -0,0 +1,44 @@
+  "description": "This is simple task manager that you can create to-do lists for projects.",
+  "main": "index.js",
+  "scripts": {
+    "start:dev": "webpack serve --open",
```
`"build": "webpack --mode production",` this would help to run the `npm run build command` for deployment

smcommits committed:
Add build and start scripts to package.json

The build script bundles code using Webpack in production
The start script handles initialization of webpack-dev-server

Forison reviewed (changes requested):
## Status::Change Required❌ 
## Somethings seems to be wrong with the update you made, The application is not longer responding, even the default project is not showing. I get this in my console🙂 both in development and production.
![Screenshot from 2021-03-17 15-09-39](https://user-images.githubusercontent.com/29695661/111524780-3bb91800-8733-11eb-90e6-203c3597168c.png)

Let us address this and submit another review.

Regards
Addo Forison

smcommits commented:
@Forison Hey! Thanks for the feedback. I suspect that there is an issue with your localStorage as the application runs fine on multiple machines I have tested. Can you please try to clean the localStorage and run the application again? Please feel free to let me know if the issue is integral to the application.

NtwaliHeritier reviewed (approved):
### **STATUS: APPROVED✔✔✔**

Great work making the required changes👏👏👏
You can merge and move on👌

Happy coding💻💻
